sources/api/openapi.yaml

```diff
@@ -324,3 +324,111 @@ paths:
           description: "Server error"
+
+  /actions/refresh-update:
+    post:
+      summary: "Query update repository and refresh list of updates"
+      operationId: "refresh_update"
+      responses:
+        204:
+          description: "Successful request"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/prepare-update:
+    post:
+      summary: "Downloads the chosen update and write the update image to the inactive partition"
+      operationId: "prepare_update"
+      responses:
+        204:
+          description: "Successful request"
+        404:
+          description: "Chosen update does not exist"
+        409:
+          description: "Action not allowed according to current update state"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/activate-update:
+    post:
+      summary: "Marks the inactive partition for boot to apply update"
+      operationId: "activate_update"
+      responses:
+        204:
+          description: "Successfully activated update"
+        404:
+          description: "No update image applied to inactive partition"
+        409:
+          description: "Action not allowed according to current update state"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /actions/deactivate-update:
+    post:
+      summary: "Deactivates update by rolling back actions done by 'activate update'"
+      operationId: "deactivate-update"
+      responses:
+        204:
+          description: "Successfully deactivated update"
+        404:
+          description: "No update image applied to inactive partition"
+        409:
+          description: "Action not allowed according to current update state"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /updates/status:
+    get:
+      summary: "Get update status"
+      operationId: "get_update_status"
+      responses:
+        200:
+          description: "Successful request"
+          content:
+            application/json:
+              schema:
+                $ref: "UpdateStatus"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /updates/available:
+    get:
+      summary: "Get details about all available updates"
+      operationId: "get_available_updates"
+      responses:
+        200:
+          description: "Successful request"
+          content:
+            application/json:
+              schema:
+                $ref: "Updates"
+        500:
+          description: "Server error"
+        423:
+          description: "Update lock held. Try again in a moment"
+
+  /updates/version-info/{version}:
```

Comment:
I'd be tempted to cut these from scope in the first pass. We really just need to know that there's an update that could be prepared, not exactly which one it is.

Per @tjkirch's feedback, maybe we need a list but I'm not sure we need any info about details yet.

Reply:
I do think we could wait and do a bit more design on the detailed APIs.  One thing we'll be missing is the wave information; you could see the list of available update versions but not understand that you can't actually update into one yet without ignoring waves.  I think that's an acceptable temporary state until we're sure of what the information should look like.

Reply:
I'll go ahead and remove these API paths and associated stuff from `thar-be-updates` and keep `available_updates` field as a simple list of versions. Does that sound OK?